content/docs/user-guide/scripting/script-canvas/programmer-guide/behavior-context.md

```diff
@@ -4,91 +4,241 @@ description: Learn about the important relationship between Script Canvas and th
-    ->Handler<BehaviorLightComponentNotificationBusHandler>();
+// Turn light on. Returns true if the light was successfully turned on.
+bool TurnOnLight();
+
+// Turn light off. Returns true if the light was successfully turned off.
+bool TurnOffLight();
+
+// Toggle light state.
+void ToggleLight();
 ```
 
-Notification buses are also known as *event handlers*. You can use these event handlers on a component to respond to the events that happen to the component. As part of reflection to the behavior context, the preceding code specifies that the `BehaviorLightComponentNotificationBusHandler` handles events for the Light component.
+These events are part of the `LightComponentRequestBus`. Their behavior is implemented by `LightComponent`.
+
+```cpp
+bool LightComponent::TurnOnLight()
+{
+    bool success = m_light.TurnOn();
+    if (success)
+    {
+        LightComponentNotificationBus::Event(GetEntityId(), &LightComponentNotifications::LightTurnedOn);
+    }
+    return success;
+}
+
+bool LightComponent::TurnOffLight()
+{
+    bool success = m_light.TurnOff();
+    if (success)
+    {
+        LightComponentNotificationBus::Event(GetEntityId(), &LightComponentNotifications::LightTurnedOff);
+    }
+    return success;
+}
+
+void LightComponent::ToggleLight()
+{
+    if (m_light.IsOn())
+    {
+        TurnOffLight();
+    }
+    else
+    {
+        TurnOnLight();
+    }
+}
+```
 
-The following code shows the binding for the `BehaviorLightComponentNotificationBusHandler` and defines two events: `LightTurnedOn` and `LightTurnedOff`.
+To make these events accessible for scripting, their methods must be reflected to the behavior context. This is done in `LightComponent::Reflect`.
+
+```cpp
+if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+{
+    behaviorContext->EBus<LightComponentRequestBus>("Light", "LightComponentRequestBus")
+        ->Attribute(AZ::Script::Attributes::Category, "Rendering")
+        ->Event("TurnOn", &LightComponentRequestBus::Events::TurnOnLight)
+        ->Event("TurnOff", &LightComponentRequestBus::Events::TurnOffLight)
+        ->Event("Toggle", &LightComponentRequestBus::Events::ToggleLight);
+}
+```
+
+When Script Canvas examines the behavior context, it finds these events and automatically generates the corresponding nodes.
+
+![Light component nodes in Script Canvas](/images/user-guide/scripting/script-canvas/behavior-context-ebus-request-light-nodes.png)
+
+#### EBus events and EntityID
+
+EBuses communicate with an entity's components. To do this, it needs an address. All component EBuses derive from `AZ::ComponentBus`, which is addressable by an ID of the type `AZ::EntityId`. For this reason, all nodes from a component EBus have a data pin for an `EntityID`. The presence of `Self` in the `EntityID` field refers to the `EntityID` of the entity that owns the Script Canvas graph. However, this ID can be assigned to another entity, or even changed to an invalid entity ID.
```

Comment:
```suggestion
EBuses communicate with an entity's components. To do this, it needs an address. All component EBuses derive from `AZ::ComponentBus`, which is addressable by an ID of the type `AZ::EntityId`. For this reason, all nodes from a component EBus have a data pin for an `EntityID`. The presence of `Self` in the `EntityID` field refers to the `EntityID` of the entity that owns the Script Canvas graph. However, this ID can be assigned to another entity, or even changed to an invalid entityId.
```
Again, not sure if a correction is appropriate here or not.

Reply:
Updated comment with some additional style formatting.